src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -0,0 +1,1349 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ControlCore.h"
+#include <argb.h>
+#include <DefaultSettings.h>
+#include <unicode.hpp>
+#include <Utf16Parser.hpp>
+#include <Utils.h>
+#include <WinUser.h>
+#include <LibraryResources.h>
+#include "../../types/inc/GlyphWidth.hpp"
+#include "../../types/inc/Utils.hpp"
+#include "../../buffer/out/search.h"
+
+#include "ControlCore.g.cpp"
+
+using namespace ::Microsoft::Console::Types;
+using namespace ::Microsoft::Console::VirtualTerminal;
+using namespace ::Microsoft::Terminal::Core;
+using namespace winrt::Windows::Graphics::Display;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::ApplicationModel::DataTransfer;
+
+namespace winrt::Microsoft::Terminal::Control::implementation
+{
+    // Helper static function to ensure that all ambiguous-width glyphs are reported as narrow.
+    // See microsoft/terminal#2066 for more info.
+    static bool _IsGlyphWideForceNarrowFallback(const std::wstring_view /* glyph */)
+    {
+        return false; // glyph is not wide.
+    }
+
+    static bool _EnsureStaticInitialization()
+    {
+        // use C++11 magic statics to make sure we only do this once.
+        static bool initialized = []() {
+            // *** THIS IS A SINGLETON ***
+            SetGlyphWidthFallback(_IsGlyphWideForceNarrowFallback);
+
+            return true;
+        }();
+        return initialized;
+    }
+
+    ControlCore::ControlCore(IControlSettings settings,
+                             TerminalConnection::ITerminalConnection connection) :
+        _connection{ connection },
+        _settings{ settings },
+        _desiredFont{ DEFAULT_FONT_FACE, 0, DEFAULT_FONT_WEIGHT, { 0, DEFAULT_FONT_SIZE }, CP_UTF8 },
+        _actualFont{ DEFAULT_FONT_FACE, 0, DEFAULT_FONT_WEIGHT, { 0, DEFAULT_FONT_SIZE }, CP_UTF8, false }
+    {
+        _EnsureStaticInitialization();
+
+        _terminal = std::make_unique<::Microsoft::Terminal::Core::Terminal>();

```

Comment:
this is going to seem very weird to a person who didn't watch this evolve

```
UWP Control─┬─────►ControlInteractivity──────────┐
      │     │                                    ▼
      │     │                              ┌──►TerminalCore
      │     └─────►ControlCore─────────────┘        │
      │                   │                         │
      │                   │                         ▼
      └───────────────────┴───────────────────►Renderer
```

core, core, core, core, core

Reply:
wait uh, isn't it

```
UWP Control─┬─────►ControlInteractivity
            │             │                      
            │             ▼                
            └─────►ControlCore────────────────►TerminalCore     
                          │                         │
                          │                         ▼
                          └───────────────────►Renderer
```

Reply:
Something like that. I hope so.